src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -282,14 +282,24 @@ protected virtual IHostnameVerifier GetSSLHostnameVerifier (HttpsURLConnection c
 				if (ReadTimeout != TimeSpan.Zero)
 					java_connection.ReadTimeout = checked ((int)ReadTimeout.TotalMilliseconds);
 
-				HttpURLConnection httpConnection = await SetupRequestInternal (request, java_connection).ConfigureAwait (continueOnCapturedContext: false);;
-				HttpResponseMessage response = await ProcessRequest (request, java_url, httpConnection, cancellationToken, redirectState).ConfigureAwait (continueOnCapturedContext: false);;
-				if (response != null)
-					return response;
-
-				if (redirectState.NewUrl == null)
-					throw new InvalidOperationException ("Request redirected but no new URI specified");
-				request.Method = redirectState.Method;
+				try {
+					HttpURLConnection httpConnection = await SetupRequestInternal (request, java_connection).ConfigureAwait (continueOnCapturedContext: false);
+					HttpResponseMessage response = await ProcessRequest (request, java_url, httpConnection, cancellationToken, redirectState).ConfigureAwait (continueOnCapturedContext: false);
+					if (response != null)
+						return response;
+
+					if (redirectState.NewUrl == null)
+						throw new InvalidOperationException ("Request redirected but no new URI specified");
+					request.Method = redirectState.Method;
+				} catch (Java.Net.SocketTimeoutException ex) {
```

Comment:
As @brendanzagaeski pointed out, these `catch` blocks should use `when (JNIEnv.ShouldWrapJavaException (ex))`, as was done in 60363ef45.